### PR TITLE
Add group endpoint tests

### PR DESCRIPTION
**Pull request by jharini**

### CODE REVIEW

File: tests/test_group_api.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+
+@eventually(5,1, {fusillade.errors.FusilladeException})
+def tearDownModule():
+    cleanup_directory(directory._dir_arn)
+    if old_directory_name:
+        os.environ["FUSILLADE_DIR"] = old_directory_name
+
+
+class TestUserApi(unittest.TestCase):
```
Comment (Bento007) [inline tests/test_group_api.py]: change to TesGroupAPI

File: tests/test_user_api.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ def test_put_new_user(self):
                 }
             },
             {
-                'name': f'400 returned when creating a user that already exists',
+                'name': f'500 returned when creating a user that already exists',
```
Comment (Bento007) [inline tests/test_user_api.py]: we should never return a 500 intentionally. What is causing this error?

File: tests/test_group_api.py (hunk 1 above)
Comment (jharini) [reply tests/test_group_api.py]: done

File: tests/test_user_api.py (hunk 2 above)
Comment (jharini) [reply tests/test_user_api.py]: Adding a bug to track this issue

Comment (Bento007) [reply tests/test_user_api.py]: add a link to the bug here

**Pull request merged**